sdk/storage/Azure.Storage.Blobs.Cryptography/src/EncryptedBlobClient.cs

```diff
@@ -0,0 +1,330 @@
+           HttpPipelinePolicy policy,
+           HttpPipelinePosition position = HttpPipelinePosition.PerCall)
+        {
+            if (options == default)
+            {
+                options = new BlobClientOptions();
+            }
+            options.AddPolicy(policy, position);
+            return options;
+        }
+        #endregion ctors
+
+        /// <summary>
+        /// This method performs a transform on the content stream for uploads. It is a no-op by default.
+        /// </summary>
+        /// <param name="content">Content to transform.</param>
+        /// <param name="metadata">Content metadata to transform.</param>
+        /// <returns>Transformed content stream.</returns>
+        internal protected override (Stream, Metadata) TransformContent(Stream content, Metadata metadata)
+        {
+            metadata ??= new Dictionary<string, string>();
+
+            var (encryptionStream, encryptionData) = EncryptStream(content);
+
+            using (var stream = new MemoryStream())
+            {
+                var serializer = new DataContractJsonSerializer(typeof(EncryptionData));
+                serializer.WriteObject(stream, encryptionData);
+                stream.Position = 0;
+                metadata.Add(EncryptionConstants.ENCRYPTION_DATA_KEY, new StreamReader(stream).ReadToEnd());
+            }
+
+            return (encryptionStream, metadata);
+        }
+
+        private (Stream, EncryptionData) EncryptStream(Stream plaintext)
+        {
+            var generatedKey = CreateKey(EncryptionConstants.ENCRYPTION_KEY_SIZE_BITS);
+
+            using (AesCryptoServiceProvider aesProvider = new AesCryptoServiceProvider() { Key = generatedKey })
+            {
+
+                var encryptionData = new EncryptionData()
+                {
+                    EncryptionMode = "FullBlob",
+                    ContentEncryptionIV = aesProvider.IV,
+                    EncryptionAgent = new EncryptionAgent()
+                    {
+                        EncryptionAlgorithm = ClientsideEncryptionAlgorithm.AES_CBC_256,
+                        Protocol = EncryptionConstants.ENCRYPTION_PROTOCOL_V1
+                    },
+                    KeyWrappingMetadata = new Dictionary<string, string>()
+                    {
+                        { EncryptionConstants.AGENT_METADATA_KEY, EncryptionConstants.AGENT_METADATA_VALUE }
+                    },
+                    WrappedContentKey = new WrappedKey()
+                    {
+                        Algorithm = null, // TODO
+                        EncryptedKey = this.KeyWrapper.WrapKey(KeyWrapAlgorithm, generatedKey).ToArray(),
+                        KeyId = this.KeyWrapper.KeyId
+                    }
+                };
+
+                return (
+                    new RollingBufferStream(
+                        new CryptoStream(plaintext, aesProvider.CreateEncryptor(), CryptoStreamMode.Read), 10 * Constants.MB,
```

Comment:
Consider moving to Encryption constants, 10 * Constants.MB as something along the lines of DefaultLengthLimitingStreamSize, and 16 as BlockSize.